packages/app-shell-vite-plugin/src/vite-configuration-processor-plugin.ts

```diff
@@ -162,3 +155,24 @@ export default function processConfiguration(
     },
   };
 }
+
+/** strips the `<script>` tags with bare `sharedDependencies` modules from the `html` string */
+function stripSharedDependencies(html: string) {
+  const sharedDeps = sharedDependencies
+    .map((dep) => dep.moduleId)
+    // escape the dependencies
+    .map((dep) => dep.replace(/[.*+?^${}()|[\]\\]/g, "\\$&"))
+    .join("|");
+
+  const sharedDepsRegex = new RegExp(
+    `<script type="module"[^>]+src="(${sharedDeps})"[^>]*><\\/script>`,
+    "g",
+  );
```

Comment:
Don't understand the trouble to only catch shared dependencies.
The previous code had it right, in that regard, imo.
The syntax for a script tag whose `src` is a bare module identifier is not supported at all and is used only as hack for an older browser that we do not support.

Reply:
Well, the previous code wasn't working on dev mode because it was removing more than it should. I believe the culprit was a `"/@vite/client"`, which _looks like_ both a path & "bare" specifier at the same time.
And so I opted for a more conservative approach.